opsdroid/connector/matrix/connector.py

```diff
@@ -111,100 +109,115 @@ def filter_json(self):
             "room": {"account_data": {"types": []}, "ephemeral": {"types": []}},
         }
 
-    async def make_filter(self, api):
+    async def make_filter(self, api, fjson):
         """Make a filter on the server for future syncs."""
-        fjson = self.filter_json
-        resp = await api.create_filter(user_id=self.mxid, filter_params=fjson)
-        return resp["filter_id"]
+        path = "/_matrix/client/r0/user/" + self.mxid + "/filter"
+        headers = {"Authorization": "Bearer " + api.token}
+        resp = await api.send(method="post", path=path, data=fjson, headers=headers)
+
+        resp_json = await resp.json()
+        return resp_json["filter_id"]
 
     async def connect(self):
         """Create connection object with chat library."""
-        session = aiohttp.ClientSession(trust_env=True)
-        mapi = AsyncHTTPAPI(self.homeserver, session)
+        config = nio.ClientConfig(encryption_enabled=False, pickle_key="")
+        mapi = nio.AsyncClient(self.homeserver, self.mxid, config=config)
 
-        self.session = session
         login_response = await mapi.login(
-            "m.login.password", user=self.mxid, password=self.password
+            password=self.password, device_name="opsdroid"
         )
-        mapi.token = login_response["access_token"]
+        if isinstance(login_response, nio.LoginError):
+            print("Error while logging in: " + login_response.message)
+
+        mapi.token = login_response.access_token
         mapi.sync_token = None
 
         for roomname, room in self.rooms.items():
-            response = await mapi.join_room(room["alias"])
-            self.room_ids[roomname] = response["room_id"]
+            response = await mapi.join(room["alias"])
+            if isinstance(response, nio.JoinError):
+                print("Error while joining room: " + roomname)
+            else:
+                self.room_ids[roomname] = response.room_id
+
         self.connection = mapi
 
         # Create a filter now, saves time on each later sync
-        self.filter_id = await self.make_filter(mapi)
+        # self.filter_id = await self.make_filter(mapi, self.filter_json)
+
+        # first_filter_id = await self.make_filter(mapi, '{ "room": { "timeline" : { "limit" : 1 } } }')
+        # print(first_filter_id)
```

Comment:
Is there a reason for removing the filter? I think it's important to only get events that we can handle.

Reply:
I'm having trouble to use the predefined filters with their IDs because nio always returns a "Invalid filter ID". The homeserver accepts the generated filter IDs. Maybe its a bug in nio. Should I just send the filters with every sync request or wait until the problem is solved?

Reply:
I'm unable to reproduce this error. Instead there was a problem with the `filter_json` property, it needs to be serialized into json in this case, passing it directly results in a bad request. Maybe that's why the filter ID isn't being recognized? because it works properly when the filter is serialized.

Reply:
I also had the problem with `filter_json` but ignored it for the moment because the serialized json made a problem in the listen function. I tried a lot with the first filter but always ran in the "Invalid filter ID" response from the homeserver. How did you serialize the json?

Reply:
`json.dumps()` worked for me and I'm not getting "invalid filter ID" with `first_filter_id` either